modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/DataSet.js

```diff
@@ -289,16 +284,37 @@ const DataSet = ({
 
 	useEffect(() => {
 		setDataLoading(true);
+		setApiError(null);
 
-		requestData().then((data) => {
+		requestData().then(({data, ok, status}) => {
 			if (isMounted()) {
-				updateDataSetItems(data);
-
+				if (!ok) {
+					setApiError({data, status});
+				}
+				else {
+					updateDataSetItems(data);
+				}
 				setDataLoading(false);
 			}
 		});
 	}, [isMounted, requestData, setDataLoading]);
 
+	useEffect(() => {
+		if (isMounted() && apiError) {
+			const {data, status: statusCode} = apiError;
+
+			const apiErrorMessage = `${data.status}, ${data.title}`;
+
+			logError(apiErrorMessage);
+
+			openToast({
+				message: apiErrorMessage,
+				title: `${Liferay.Language.get('error')} ${statusCode}`,
+				type: 'danger',
+			});
+		}
+	}, [apiError, isMounted]);
+
```

Comment:
Do we need `useEffect` here, with `apiError` as a state variable? It seems to me that this code can be simplified to be inline in `requestData().then(...`, or in a helper function if you want to 'buy space'.